extensions/poetry/main.ts

```diff
@@ -0,0 +1,98 @@
+import { PhylumApi } from "phylum"
+import { red, green, yellow } from "https://deno.land/std@0.150.0/fmt/colors.ts";
+
+class FileBackup {
+  readonly fileName: string
+  readonly fileContent: string | null
+
+  constructor(fileName: string) {
+    this.fileName = fileName
+    this.fileContent = null
+  }
+
+  async backup() {
+    try {
+      this.fileContent = await Deno.readTextFile(this.fileName)
+    } catch (e) { }
+  }
+
+  async restoreOrDelete() {
+    try {
+      if (this.fileContent != null) {
+        await Deno.writeTextFile(this.fileName, this.fileContent)
+      } else {
+        await Deno.remove(this.fileName)
+      }
+    } catch (e) { }
+  }
+}
+
+// Parse the output of `poetry` subcommands that support the `--dry-run` flag,
+// and submit the packages.
+//
+// Add the `-vvv` flags to get a detailed report of the dependency resolution
+// process. This way, we can track the actual lockfile changes, as specifying
+// `--dry-run` only would not output anything in combination with flags such
+// as `--lock` that do not perform the actual operations.
+async function poetryCheckDryRun(subcommand: string, args: string[]): boolean {
+  // Read and backup the current poetry lockfile contents.
+  const lockfileBackup = new FileBackup('poetry.lock')
+  const manifestBackup = new FileBackup('pyproject.toml')
+
+  await lockfileBackup.backup()
+  await manifestBackup.backup()
+
+  let process = Deno.run({
+    cmd: ['poetry', subcommand, '-vvv', '-n', '--dry-run', ...args.map(s => s.toString())],
+    stdout: 'piped',
+    stderr: 'piped',
+  })
+
+  await process.status()
+  await process.close()
```

Comment:
Is it necessary to await close when already awaiting status? A process cannot return a status without closing, right? IIRC Deno's API docs also don't call close after status.

Reply:
Let me check that out -- I added this because I observed some process hanging at some point, but I'm not confident I haven't solved that by other means.

Reply:
Considering how processes generally work, that would really surprise me. So please do let me know since it would be relevant for our other ecosystem extensions too.